src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -139,166 +88,66 @@ export class QuerySuggestPreview extends Component implements IComponentBindings
-    previousQueryOptions.q = this.currentlyRenderedSuggestion;
+    previousQueryOptions.q = suggestionText;
     previousQueryOptions.numberOfResults = this.options.numberOfPreviewResults;
+    this.logShowQuerySuggestPreview();
     const results = await this.queryController.getEndpoint().search(previousQueryOptions);
-    $$(this.previewContainer).empty();
-    this.currentlyDisplayedResults = [];
     if (!results) {
-      return;
+      return [];
     }
-    this.buildPreviewHeader(this.currentlyRenderedSuggestion);
-    this.buildResultsPreview(results);
+    return this.buildResultsPreview(suggestionText, results);
   }
 
-  private async buildResultsPreview(results: IQueryResults) {
-    const resultsContainer = this.buildResultsContainer();
-    this.previewContainer.appendChild(resultsContainer);
-    this.setupRenderer(resultsContainer);
-    const buildResults = await this.templateToHtml.buildResults(results, 'preview', this.currentlyDisplayedResults);
+  private async buildResultsPreview(suggestionText: string, results: IQueryResults) {
+    const buildResults = await this.templateToHtml.buildResults(results, 'preview', []);
     if (!(buildResults.length > 0)) {
-      return;
+      return [];
     }
-    this.updateResultElement(buildResults);
-    this.addOnClickListener(buildResults);
-    this.renderer.renderResults(buildResults, true, result => {});
+    return buildResults.map((element, index) => this.buildResultPreview(suggestionText, element, index));
   }
 
-  private updateResultElement(elements: HTMLElement[]) {
-    const resultAvailableSpace = elements.length === 4 ? '50%' : '33%';
-    elements.forEach(element => {
-      $$(element).addClass('coveo-preview-selectable');
-
-      $$(element).on('keyboardSelect', () => {
-        this.handleSelect(element);
-      });
-
-      this.updateResultPerRow(element, resultAvailableSpace);
-    });
+  private buildResultPreview(suggestionText: string, element: HTMLElement, rank: number): ISearchResultPreview {
+    $$(element).addClass('coveo-preview-selectable');
+    return {
+      element,
+      onSelect: () => this.handleSelect(suggestionText, element, rank)
+    };
   }
 
-  private handleSelect(element: HTMLElement) {
+  private handleSelect(suggestionText: string, element: HTMLElement, rank: number) {
+    this.logClickQuerySuggestPreview(suggestionText, rank, element);
     element.click();
     const link = $$(element).find(`.${Component.computeCssClassNameForType('ResultLink')}`);
     if (link) {
       const resultLink = <ResultLink>Component.get(link);
-      resultLink.openLinkAsConfigured() || resultLink.openLink();
+      resultLink.openLinkAsConfigured();
+      resultLink.openLink();
     }
   }
 
-  private addOnClickListener(results: HTMLElement[]) {
-    results.forEach(result => {
-      const rank = results.indexOf(result);
-      this.bind.on(result, 'click', (e: MouseEvent) => {
-        this.handleOnClick(e, result, rank);
-      });
```

Comment:
This is now bound by `SuggestionsManager.updateSearchResultPreviews` and handled by `QuerySuggestPreview.handleSelect`